src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -542,6 +542,14 @@ public final boolean referencesUserType(ByteBuffer name)
         return false;
     }
 
+    /**
+     * Whether this type is or contains any UDT.
+     */
+    public final boolean referencesUserTypes()
+    {
+        return isUDT() || subTypes().stream().anyMatch(AbstractType::referencesUserTypes);
+    }
+
```

Comment:
This is pulled in transitively as it is used in changes made to  `ColumnMetadata` and `DroppedColumn`.